core/src/main/java/org/elasticsearch/monitor/process/JmxProcessProbe.java

```diff
@@ -86,16 +118,179 @@ public JmxProcessProbe(Settings settings) {
         super(settings);
     }
 
+    /**
+     * @return the process identifier (PID), as returned
+     * by the RuntimeMXBean bean through JVMInfo class.
+     */
+    @Override
+    public Long pid() {
+        return jvmInfo().pid();
+    }
+
+    /**
+     * @return the maximum number of file descriptors,
+     * as returned by the OperatingSystemMXBean bean.
+     */
+    @Override
+    public Long maxFileDescriptor() {
+        try {
+            if (getMaxFileDescriptorCountField != null) {
+                return (Long) getMaxFileDescriptorCountField.invoke(osMxBean);
+            }
+        } catch (Exception e) {
+            logger.trace("exception when accessing OperatingSystemMXBean.getMaxFileDescriptorCountField()");
+        }
+        return null;
+    }
+
+    /**
+     * @return the number of open file descriptors,
+     * as returned by the OperatingSystemMXBean bean.
+     */
+    @Override
+    public Long openFileDescriptor() {
+        try {
+            if (getOpenFileDescriptorCountField != null) {
+                return (Long) getOpenFileDescriptorCountField.invoke(osMxBean);
+            }
+        } catch (Exception e) {
+            logger.trace("exception when accessing OperatingSystemMXBean.getOpenFileDescriptorCountField()");
+        }
+        return null;
+    }
+
+    /**
+     * @return the CPU load the current JVM is taking
+     */
+    @Override
+    public Short processCpuLoad() {
+        try {
+            if (getProcessCpuLoad != null) {
+                double load = (double) getProcessCpuLoad.invoke(osMxBean);
+                if (load >= 0) {
+                    return (short) (load * 100);
+                }
+            }
+        } catch (Exception e) {
+            logger.trace("exception when accessing OperatingSystemMXBean.getProcessCpuLoad()");
+        }
+        return null;
+    }
+
+    /**
+     * @return the CPU time used by the process on which the Java virtual machine is running in millis
+     */
+    @Override
+    public Long processCpuTime() {
+        try {
+            if (getProcessCpuTime != null) {
+                long time = (long) getProcessCpuTime.invoke(osMxBean);
+                if (time >= 0) {
+                    return (time / 1_000_000L);
+                }
+            }
+        } catch (Exception e) {
+            logger.trace("exception when accessing OperatingSystemMXBean.getProcessCpuTime()");
+        }
+        return null;
+    }
+
+    @Override
+    public Long processSystemTime() {
+        if (!threadMxBean.isThreadCpuTimeSupported()) {
+            throw new UnsupportedOperationException("JmxProcessProbe does not support process user time");
+        }
```

Comment:
No. This method is documented as the flag to indicate if CPU time is supported, I think it makes sense to use it here before requesting value